documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,932 @@
+
+#### Example of closing a client in the v2 SDK
+
+```typescript
+// Shutdown and clean up.
+const stopped = once(client, Mqtt5Client.STOPPED);
+client.stop();
+await stopped;
+
+// Release any native resources associated with the client.
+client.close();
+```
+
+
+### Reconnects
+
+In the v1 SDK, the `reconnectPeriod` option enables reconnection. The default value is 1000 ms which means it will try to
+reconnect one second after losing the connection.
+
+The v2 SDK attempts to reconnect automatically until connection succeeds or `client.stop()` is called. The reconnection parameters,
+such as min/max delays and [jitter modes](https://aws.github.io/aws-iot-device-sdk-js-v2/node/classes/iot.AwsIotMqtt5ClientConfigBuilder.html#withRetryJitterMode),
+are configurable through [AwsIotMqtt5ClientConfigBuilder](https://aws.github.io/aws-iot-device-sdk-js-v2/node/classes/iot.AwsIotMqtt5ClientConfigBuilder.html).
+
+
+#### Example of tweaking reconnection settings in the v1 SDK
+
+```typescript
+var device = deviceModule({
+    certPath: "<certificate file>",
+    // ...
+    reconnectPeriod: 1000,          // ms
+    minimumConnectionTimeMs: 2500,  // ms
+    maximumReconnectTimeMs: 10000   // ms
+});
+```
+
+#### Example of tweaking reconnection settings in the v2 SDK
+
+```typescript
+configBbuilder.withMinReconnectDelayMs(2500);
+configBbuilder.withMaxReconnectDelayMs(10000);
+configBbuilder.withRetryJitterMode(mqtt5.RetryJitterMode.Full);
+
+let config : mqtt5.Mqtt5ClientConfig = configBuilder.build();
+let client : mqtt5.Mqtt5Client = new mqtt5.Mqtt5Client(config);
+```
+
+
+### Offline Operations Queue
+
+The v1 SDK provides a set of options to configure behavior when a client is offline. By default, the number of queued messages
+are not limited.
+
+The v2 SDK does not limit the number of in-flight messages. Additionally, the v2 SDK provides a way to configure which kind of
+packets will be placed into the offline queue when the client is in the disconnected state. The following code snippet
+demonstrates how to enable storing all packets except QOS0 publish packets in the offline queue on disconnect:
+
+#### Example of configuring the offline queue in the v1 SDK
+
+```typescript
+var device = deviceModule({
+    // ...
+    offlineQueueing: true,
+    offlineQueueMaxSize: 0,  // no limit
+    offlineQueueDropBehavior: "oldest",
+    drainTimeMs: 250,  // ms
+});
+```
+
+#### Example of configuring the offline queue in the v2 SDK
+
+```typescript
+configBuilder.withOfflineQueueBehavior(mqtt5.ClientOperationQueueBehavior.FailQos0PublishOnDisconnect);
+let config : mqtt5.Mqtt5ClientConfig = configBuilder.build();
+let client : mqtt5.Mqtt5Client = new mqtt5.Mqtt5Client(config);
+```
+
+> [!NOTE]
+> AWS IoT Core [limits the number of allowed operations per second](https://docs.aws.amazon.com/general/latest/gr/iot-core.html#message-broker-limits).
```

Comment:
This is the note below. If a large enough pool of queued messages are stored, on connect/reconnect they could all be processed at once causing this limit to be hit and a disconnect to occur from the broker. I believe our v2 mqtt5 client has checks in place that can be used to prevent this from occurring. a throttle to prevent sending too many messages too fast from a queue.